lib/ansible/modules/packaging/language/pip.py

```diff
@@ -503,12 +571,24 @@ def main():
         # Automatically apply -e option to extra_args when source is a VCS url. VCS
         # includes those beginning with svn+, git+, hg+ or bzr+
         has_vcs = False
+
         if name:
             for pkg in name:
                 if bool(pkg and re.match(r'(svn|git|hg|bzr)\+', pkg)):
                     has_vcs = True
                     break
 
+            distributions = _get_distributions_from_names(module, name)
+            # check invalid combination of arguments
+            if version is not None:
+                if len(distributions) > 1:
+                    module.fail_json(msg="'version' is not available for installing multi-packages.")
+                if distributions[0].has_version_specifier:
+                    module.fail_json(msg="'version' argument conflicts with version specifier provided along with a package name.\
+                                     Please keep a version specifier, but remove the 'version' argument.")
+                else:
+                    distributions[0] = Distribution(distributions[0].distribution_name, version)
```

Comment:
Why don't you assign string here? Your further loops assume it's a string and convert it for you anyway.

Reply:
There is a more further loop in check_mode that need the `distributions` to be instances of `Distribution`.